// homey-api/lib/AthomDeviceFirmwareAPI.js
'use strict';

const API = require('./API');

class AthomDeviceFirmwareAPI extends API {

  static SPECIFICATION = {
    host: 'device-firmware.athom.com',
    basePath: '/',
    operations: {},
  };

  static JSDOC_PRIVATE = true;

  /**
   * Get the location of a firmware update file.
   *
   * @param {Object} opts
   * @param {string} opts.integrity - The integrity hash (i.e unique key) of the firmware file, e.g. "sha256:abcdef1234567890..."
   * @param {boolean} [opts.noCache] - Whether to bypass the cache and fetch the firmware fresh from sourceUrl. Default: false.
   * @param {string} [opts.sourceUrl] - The original URL of the firmware file, if the cache is bypassed or missed the server can attempt to fetch the file from this URL.
   * @param {string} [opts.homeyTag] - How to tag the file in the cache (requires sourceUrl to be set, and the cache to be bypassed or missed).
   * @param {number} [opts.$timeout] - Optional timeout in milliseconds for this API call.
   * @returns {Promise<{ url: string, cache: string }>}
   */
  async getFirmwareURL({ integrity, noCache, sourceUrl, homeyTag, $timeout }) {
    const query = {};

    if (typeof this.__secret === 'string') {
      query.secret = this.__secret;
    }

    if (noCache !== undefined) {
      query.noCache = noCache;
    }

    if (sourceUrl !== undefined) {
      query.sourceUrl = sourceUrl;
    }

    if (homeyTag !== undefined) {
      query.homeyTag = homeyTag;
    }

    return this.call({
      method: 'get',
      path: `/firmware/${integrity}`,
      query,
      timeout: $timeout,
    });
  }

}

module.exports = AthomDeviceFirmwareAPI;
